### ISSUE TITLE
Fix anonymous class violation file name prepending

### ISSUE THREAD

**Issue by kylekatarnls** [phpmd/phpmd#837]
- PHPMD version: 2.9.1
- PHP Version: 7.4.10
- Installation type: composer
- Operating System / Distribution & Version: Alpine Linux in Docker

Next step for #829

### Current Behavior
```
:9 Violation
```

### Expected Behavior
```
FileName.php:9 Violation
```

### Steps To Reproduce:
Run `cleancode` rule:
```
<?php
class TestCase
{
    public function getAnonymousClass()
    {
        return new class {
            public function get(): string
            {
                return '';
            }
        };
    }
}
```

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by tvbeek** [comment]
@kylekatarnls is this already fixed in master? I tried to reproduce it but did get the filename in the error. Else I miss something and got the wrong error.

**Comment by AJenbo** [comment]
Unable to reproduce the issue in 3.x
